perma_web/frontend/lib/store.js

```diff
@@ -12,14 +13,57 @@ export const getLinksRemainingStatus = (linksRemaining, isNonpaying) => {
     return globalStore.updateLinksRemainingStatus('unlimited_paid')
 }
 
-export const getUserTypes = (isIndividual) => {
-    let types = []
-
+export const getUserTypes = (isIndividual, user) => {
     if (isIndividual) {
-        types = types.concat('individual')
+        return globalStore.updateUserTypes('individual')
+    }
+
+    let userTypes = []
+
+    const topLevelFolders = user.top_level_folders
+    const isOrgAffiliated = topLevelFolders.some(folder => !!folder.organization)
+    const isSponsored = topLevelFolders.some(folder => folder.is_sponsored_root_folder)
+
+    if (isOrgAffiliated) {
+        userTypes = userTypes.concat('orgAffiliated')
+    }
+
+    if (isSponsored) {
+        userTypes = userTypes.concat('sponsored')
+    }
+
+    if (userTypes.length) {
+        return globalStore.updateUserTypes(userTypes)
+    }
+}
+
+export const getOrganizationFolders = async () => {
+    const { data, error, errorMessage } = await useFetch('/api/v1/organizations', {
+        limit: 300,
+        order_by: 'registrar, name'
+    })
+
+    if (error) {
+        globalStore.updateFetchErrorMessage(errorMessage)
+    }
+
+    if (data?.value?.objects.length) {
+        const organizationsById = data.value.objects.reduce((acc, currentValue) => {
+            return { ...acc, [currentValue.id]: currentValue }
+        }, {})
+        globalStore.updateOrganizationFolders(data.value.objects)
+        globalStore.updateOrganizationFoldersById(organizationsById)
+    }
```

Comment:
FYI: I don't believe we currently need `organizationsById` just yet, but this was generated by the legacy code so it felt like an easy addition. I imagine it might be more useful in the future, when more functions are modernized.

Reply:
Yeah, I bet we can come up with a single data structure that serves both purposes, rather than using up memory holding storing this data twice. I'm less familiar with the idiomatic way to do this in Javascript, but in Python, a person could use an OrderedDict do the job of both: insert objects in order into the dict, and then call `dict.values` when you just want the list of objects, without the keys. Is there something similar that would serve here?

Reply:
For sure, there are definitely ways to iterate on a dictionary like structure in JavaScript — and keep it as both a dictionary and iterate on it like an array. This isn't super common in my experience to do, which might simply be in part due to the fact that until ES2020 the order of an object wasn't guaranteed (and differed in implementation between Webkit and other browsers, I think). But it could definitely make sense!

I deleted this in c017fa75b5f6edf32916c32d332595bcce2e0316 for now, because we don't need it, and think we can easily revisit it in the future.

Reply:
> and differed in implementation between Webkit and other browsers
lol of course, isn't that always the way?

Cool, thanks yes, this sounds totally reasonable to me :-)